dashboard/app/models/email_preference.rb

```diff
@@ -38,14 +38,18 @@ def email=(value)
 
   def self.upsert!(email:, opt_in:, ip_address:, source:, form_kind:)
     email_preference = EmailPreference.find_or_initialize_by(email: email)
-    email_preference.update!(
-      email: email,
-      # Don't change opt_in to false if the record exists already.  We currently only allow user to opt out via Pardot
-      # unsubscribe link.
-      opt_in: (!email_preference.new_record? && !opt_in) ? email_preference.opt_in : opt_in,
-      ip_address: ip_address,
-      source: source,
-      form_kind: form_kind
-    )
+    # Don't change opt_in to false if a preference with opt_in = true exists already.
+    # We currently only enable a user to opt out via the Pardot unsubscribe link.
+    unless !email_preference.new_record? && !opt_in && email_preference.opt_in
```

Comment:
I always struggle with parsing this kind of logic, especially with `unless`.  Is there a chance this is the same as 

```
if email_preference.new_record? || opt_in || !email_preference.opt_in
```

And is that helpful?

Reply:
Agreed.